### ISSUE TITLE
Screen Reader does not work as expected

### ISSUE THREAD

**Issue by ANAZHTHTHS** [microsoft/vscode#146773]
Issue Type: <b>Bug</b>

I've just opened the software and it asked me if I use a screen reader (so I said yes). I use NVDA because I can see a little (using magnify) but I want to listen too. The problem is that when I am navigating with keyboard I have the cursor in a line and screen reader reads the previous line.
It is not posible for me to work like that so I wanted to adress the problem.
Thank you. 

VS Code version: Code 1.66.0 (e18005f0f1b33c29e81d732535d8c0e47cafb0b5, 2022-03-30T05:50:14.623Z)
OS version: Windows_NT x64 10.0.19044
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|AMD FX(tm)-6300 Six-Core Processor              (6 x 3900)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|7.95GB (3.31GB free)|
|Process Argv|--crash-reporter-id 5d0546a0-92a0-4be0-9516-41ebe37fb929|
|Screen Reader|yes|
|VM|0%|
</details>Extensions: none<details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
pythonvspyl392:30443607
pythontb:30283811
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
vscorecescf:30445987
pythondataviewer:30285071
vscod805:30301674
pythonvspyt200:30340761
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593:30376534
vsc1dst:30438360
pythonvs932:30410667
wslgetstarted:30449410
pythonvsnew555:30457759
vscscmwlcmt:30465135
cppdebugcf:30466690
vscaat:30438848
pynewfile477:30463512

```

</details>

**Comment by isidorn** [comment]
@ANAZHTHTHS thanks for opening this issue and sharing feedback.

I could not reproduce this. So you are saying that you turn on NVDA, and as you move the cursor NVDA always reads the previous line?

Can you reproduce this for every file? Can you reproduce if you do not use magnify? Can you reproduce with VS Code Insiders https://code.visualstudio.com/insiders/

**Comment by ANAZHTHTHS** [comment]
Thank you for your quick response. I’ve tried it again today both standard and insiders version. There is no problem with the lines any more.
The problem now is that when I am reading a line of code and I am moving to the next line with keyboard arrows, the magnifier (default magnifier for windows 10  21H2 19044.1586 x65) loses the target (like in the screenshot).
Also I have set NVDA so as to speak whatever is under the mouse cursor. But in the VSC and VSC insider it doesn’t read the code (it works perfect with the buttons).
Thank you for your time.

PS: English is not my native language so forgive me for any mistakes.

Georgios Tsagkalidis
Special Education Teacher


Στάλθηκε από την Αλληλογραφία για τα Windows

Από: Isidor Nikolic
Αποστολή: Τρίτη, 5 Απριλίου 2022 10:23 πμ
Προς: microsoft/vscode
Κοιν.: ANAZHTHTHS; Mention
Θέμα: Re: [microsoft/vscode] Screen Reader does not work as expected (Issue#146773)

@ANAZHTHTHS thanks for opening this issue and sharing feedback.
I could not reproduce this. So you are saying that you turn on NVDA, and as you move the cursor NVDA always reads the previous line?
Can you reproduce this for every file? Can you reproduce if you do not use magnify? Can you reproduce with VS Code Insiders https://code.visualstudio.com/insiders/

**Comment by ANAZHTHTHS** [comment]
I am sorry I didn’t attached the screenshot.


Στάλθηκε από την Αλληλογραφία για τα Windows

Από: Isidor Nikolic
Αποστολή: Τρίτη, 5 Απριλίου 2022 10:23 πμ
Προς: microsoft/vscode
Κοιν.: ANAZHTHTHS; Mention
Θέμα: Re: [microsoft/vscode] Screen Reader does not work as expected (Issue#146773)

@ANAZHTHTHS thanks for opening this issue and sharing feedback.
I could not reproduce this. So you are saying that you turn on NVDA, and as you move the cursor NVDA always reads the previous line?
Can you reproduce this for every file? Can you reproduce if you do not use magnify? Can you reproduce with VS Code Insiders https://code.visualstudio.com/insiders/

**Comment by isidorn** [comment]
I believe you are hitting this issue https://github.com/microsoft/vscode/issues/105558
I suggest to use the workaround for now In the meantime, set `editor.accessibilityPageSize: 1` in your vscode settings.
If you do that can you still reproduce the issue?
